src/vs/workbench/contrib/extensions/browser/extensionsActions.ts

```diff
@@ -525,12 +525,17 @@ export class InstallingLabelAction extends ExtensionAction {
 	private static readonly LABEL = localize('installing', "Installing");
 	private static readonly CLASS = `${ExtensionAction.LABEL_ACTION_CLASS} install installing`;
 
-	constructor() {
+	constructor(
+		@IExtensionService private readonly extensionService: IExtensionService,
+		@IExtensionsWorkbenchService private readonly extensionsWorkbenchService: IExtensionsWorkbenchService,
+	) {
 		super('extension.installing', InstallingLabelAction.LABEL, InstallingLabelAction.CLASS, false);
+		this._register(this.extensionService.onDidChangeExtensions(() => this.update()));
```

Comment:
> What is the reason for listening to this event?
It is how we notice when `this.extensionsWorkbenchService.isInstalling` goes false, at which point the Reload Required must appear if it is needed by this extension.

> I think we should leverage following API
I'm not clear what you mean.

Reply:
Sorry for not being clear. My recommendation is to use this API to bulk update the extensions. When user clicks on update all button, we should call this API. 

- Add an API `updateAll` to `IExtensionsWorkbenchService` that is implemented using above API
- Update All action uses this new API of `IExtensionsWorkbenchService` instead of calling install on each outdated extension

This way we do not need `isInstalling` state and actions do not need to check for this state, because bulk install will trigger install complete event for all extensions after all extensions are installed.